wgpu/backends/wgpu_native/_api.py

```diff
@@ -778,24 +808,44 @@ def _request_device(
 
         # ----- Set limits
 
+        # H: chain: WGPUChainedStruct, limits: WGPUNativeLimits
+        c_required_limits_extras = new_struct_p(
+            "WGPURequiredLimitsExtras *",
+            # not used: chain
+            # not used: limits
+        )
+        c_required_limits_extras.chain.sType = lib.WGPUSType_RequiredLimitsExtras
         # H: nextInChain: WGPUChainedStruct *, limits: WGPULimits
         c_required_limits = new_struct_p(
             "WGPURequiredLimits *",
-            # not used: nextInChain
+            nextInChain=ffi.cast("WGPUChainedStruct*", c_required_limits_extras),
             # not used: limits
         )
         c_limits = c_required_limits.limits
+        c_limits_extras = c_required_limits_extras.limits
+
+        if required_limits:
+            # Do some quick error checking
+            unknown_keys = [x for x in required_limits if x not in self._limits]
+            if unknown_keys:
+                raise KeyError(f"Unknown required limits {', '.join(unknown_keys)}")
+        else:
+            # If required_limits isn't set, set it to self._limits.  This is the same as
+            # setting it to {}, but the loop below goes just a little bit faster.
+            required_limits = required_limits or {}
 
```

Comment:
The code does something different than what the comment says it does. Also `bool(required_limits)` is  already established to be false here.

Reply:
Fixed.  My mistake.